src/java.base/share/classes/sun/util/locale/BaseLocale.java

```diff
@@ -93,7 +93,7 @@ public final class BaseLocale {
 
     // Interned BaseLocale cache
     private static final ReferencedKeySet<BaseLocale> CACHE =
-            ReferencedKeySet.create(true, ConcurrentHashMap::new);
+            ReferencedKeySet.create(true, ReferencedKeySet.concurrentHashMapSupplier());
```

Comment:
Should this supplier be in `BaseLocale` class? Otherwise `ReferencedKeySet` may end up with static suppliers for each map type?

Reply:
Maybe, though I think most potential uses of `ReferenceKeySet/-Map` will want to be using a plain `CHM` as the backing store, so keeping these next to their respective `create` methods will encourage sharing.